tests/test_internals/test_nwb_extractors.py

```diff
@@ -0,0 +1,109 @@
+from pynwb.image import ImageSeries
+from pynwb.ophys import TwoPhotonSeries, OpticalChannel, ImageSegmentation
+
+from roiextractors import NwbImagingExtractor
+from roiextractors.testing import generate_dummy_video
+from roiextractors.extraction_tools import PathType, ArrayType
+
+
+class TestNwbImagingExtractor(unittest.TestCase):
+    def setUp(self) -> None:
+        self.session_start_time = datetime.now().astimezone()
+        self.file_path = Path(mkdtemp()) / "test_nwb_imaging_extractor.nwb"
+
+        self.sampling_frequency = 30.0
+        self.num_frames = 30
+        self.rows = 50
+        self.columns = 25
+        self.num_channels = 1
+
+        self.nwbfile = NWBFile(
+            session_description="session_description",
+            identifier="file_id",
+            session_start_time=self.session_start_time,
+        )
+        self.device = self.nwbfile.create_device(name="Microscope")
+
+        channel_names = [f"channel_num_{num}" for num in range(self.num_channels)]
+        self.optical_channel_list = [
+            OpticalChannel(name=channel_name, description="description", emission_lambda=500.0)
+            for channel_name in channel_names
+        ]
+        self.video_shape = (self.num_frames, self.rows, self.columns)
+        self.image_size = (self.rows, self.columns)
+
+        self.dtype = "uint"
+        self.video = generate_dummy_video(size=self.video_shape, dtype=self.dtype)
+
+        self.imaging_plane = self.nwbfile.create_imaging_plane(
+            name="ImagingPlane",
+            optical_channel=self.optical_channel_list,
+            imaging_rate=self.sampling_frequency,
+            description="a very interesting part of the brain",
+            device=self.device,
+            excitation_lambda=600.0,
+            indicator="GFP",
+            location="the location in the brain",
+        )
+
+        # using internal data. this data will be stored inside the NWB file
+        self.image_series = TwoPhotonSeries(
+            name="TwoPhotonSeries",
+            data=self.video,
+            imaging_plane=self.imaging_plane,
+            rate=self.sampling_frequency,
+            unit="normalized amplitude",
+        )
+
+        self.nwbfile.add_acquisition(self.image_series)
+
+        with NWBHDF5IO(self.file_path, "w") as io:
+            io.write(self.nwbfile)
+
+    def test_basic_setup(self):
+
+        nwb_imaging_extractor = NwbImagingExtractor(file_path=self.file_path)
+
+        image_size = nwb_imaging_extractor.get_image_size()
+        num_frames = nwb_imaging_extractor.get_num_frames()
+        num_channels = nwb_imaging_extractor.get_num_channels()
+
+        expected_image_size = self.image_size
+        expected_num_frames = self.num_frames
+        expected_num_channels = self.num_channels
+
+        assert image_size == expected_image_size
+        assert num_frames == expected_num_frames
+        assert num_channels == expected_num_channels
+
+        frame_idxs = 0
```

Comment:
Maybe this case does not make sense, I am not sure if we should allow `frame_idx` take scalar values instead of list or arrays.

Reply:
I just checked and `TiffImagingExtractor` does the same. In a second thought my current take is that  the argument frame_idxs of `get_frames` should behave exactly like slicing of `numpy` to the extent that is possible so users use a mental model that is readily available within the scientific community. Moreover, if we do get that we can abstract this away with [__getitem__ ](https://docs.python.org/3/reference/datamodel.html#object.__getitem__) to support slicing notation.

Reply:
> Maybe this case does not make sense, I am not sure if we should allow frame_idx take scalar values instead of list or arrays.

It's mostly because `frame_idxs=[single_idx]` is more awkward than `frame_idxs=single_idx`.

Reply:
> `get_frames` should behave exactly like slicing of numpy

Right, and you can slice a numpy array with a single scalar as well as an `ArrayType` of multiple indexes.

Reply:
See:
https://github.com/catalystneuro/roiextractors/issues/147